[Skia][GTK][WPE][Win] Extract DragImageSkia

yury-s opened the pull request:
#### adefaa9ed9f844791dbd50548f3f744d84d06974
<pre>
[Skia][GTK][WPE][Win] Extract DragImageSkia
<a href="https://bugs.webkit.org/show_bug.cgi?id=293657">https://bugs.webkit.org/show_bug.cgi?id=293657</a>

Reviewed by Don Olmstead and Carlos Garcia Campos.

Extracted Skia implementation of DragImage from DragImageGtk into DragImageSkia
and started using it on Windows and WPE.

* Source/WebCore/PlatformGTK.cmake:
* Source/WebCore/PlatformWPE.cmake:
* Source/WebCore/PlatformWin.cmake:
* Source/WebCore/SourcesGTK.txt:
* Source/WebCore/platform/DragImage.h:
* Source/WebCore/platform/gtk/DragImageGtkCairo.cpp: Copied from Source/WebCore/platform/gtk/DragImageGtk.cpp.
Only Cairo-specific bits are left in the file, so it was renamed.

(WebCore::dragImageSize):
(WebCore::deleteDragImage):
(WebCore::scaleDragImage):
(WebCore::dissolveDragImageToFraction):
(WebCore::createDragImageFromImage):
(WebCore::createDragImageIconForCachedImageFilename):
(WebCore::createDragImageForLink):
(WebCore::createDragImageForColor):
* Source/WebCore/platform/mediastream/libwebrtc/gstreamer/GStreamerVideoEncoderFactory.cpp: Non-unified build fix.

* Source/WebCore/platform/skia/DragImageSkia.cpp: Renamed from Source/WebCore/platform/gtk/DragImageGtk.cpp.
Skia specific code from DragImageGtk, verbatim.

(WebCore::dragImageSize):
(WebCore::deleteDragImage):
(WebCore::scaleDragImage):
(WebCore::dissolveDragImageToFraction):
(WebCore::createDragImageFromImage):
(WebCore::createDragImageIconForCachedImageFilename):
(WebCore::createDragImageForLink):
(WebCore::createDragImageForColor):
* Source/WebCore/platform/win/DragImageWin.cpp:
Merged remaining logic into DragImageWinCairo.cpp as it is all specific to Cairo.

* Source/WebCore/platform/cairo/DragImageCairo.cpp: Copied from Source/WebCore/platform/gtk/DragImageGtk.cpp.
Droped GTK3 specific code and moved the rest of cairo code to cairo directory.

Canonical link: <a href="https://commits.webkit.org/295695@main">https://commits.webkit.org/295695@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/63eb39a41526cba9875d9850563b02004ecac861

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/105403 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/25116 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/15542 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/110608 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/56040 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/107444 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/25580 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/33654 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/80055 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/108409 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/19961 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/95121 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/60364 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/19695 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/13208 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/55446 "Built successfully") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/89424 "Found 1 new API test failure: TestWebKitAPI.WebKit.QuotaDelegate (failure)") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/13249 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/113338 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/32549 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/23994 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/89131 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/32913 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/91347 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/88777 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22732 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/33655 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/11452 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/28013 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/32477 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/37901 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/32235 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/35583 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/33820 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos reviewed (changes requested):
Looks good, but I would just split into cairo and skia, not gtk-cairo and skia. Don't worry about the gtk3 code in DragImageGtk.cpp, I'm not even sure that's useful nowadays. Also note that WPE doesn't support drag and drop and it's always built with drag disabled.

yury-s committed:
[Skia][GTK][WPE][Win] Extract DragImageSkia
https://bugs.webkit.org/show_bug.cgi?id=293657

Reviewed by Don Olmstead and Carlos Garcia Campos.

Extracted Skia implementation of DragImage from DragImageGtk into DragImageSkia
and started using it on Windows and WPE.

* Source/WebCore/PlatformGTK.cmake:
* Source/WebCore/PlatformWPE.cmake:
* Source/WebCore/PlatformWin.cmake:
* Source/WebCore/SourcesGTK.txt:
* Source/WebCore/platform/DragImage.h:
* Source/WebCore/platform/gtk/DragImageGtkCairo.cpp: Copied from Source/WebCore/platform/gtk/DragImageGtk.cpp.
Only Cairo-specific bits are left in the file, so it was renamed.

(WebCore::dragImageSize):
(WebCore::deleteDragImage):
(WebCore::scaleDragImage):
(WebCore::dissolveDragImageToFraction):
(WebCore::createDragImageFromImage):
(WebCore::createDragImageIconForCachedImageFilename):
(WebCore::createDragImageForLink):
(WebCore::createDragImageForColor):
* Source/WebCore/platform/mediastream/libwebrtc/gstreamer/GStreamerVideoEncoderFactory.cpp: Non-unified build fix.

* Source/WebCore/platform/skia/DragImageSkia.cpp: Renamed from Source/WebCore/platform/gtk/DragImageGtk.cpp.
Skia specific code from DragImageGtk, verbatim.

(WebCore::dragImageSize):
(WebCore::deleteDragImage):
(WebCore::scaleDragImage):
(WebCore::dissolveDragImageToFraction):
(WebCore::createDragImageFromImage):
(WebCore::createDragImageIconForCachedImageFilename):
(WebCore::createDragImageForLink):
(WebCore::createDragImageForColor):
* Source/WebCore/platform/win/DragImageWin.cpp:
Merged remaining logic into DragImageWinCairo.cpp as it is all specific to Cairo.

* Source/WebCore/platform/cairo/DragImageCairo.cpp: Copied from Source/WebCore/platform/gtk/DragImageGtk.cpp.
Droped GTK3 specific code and moved the rest of cairo code to cairo directory.

Canonical link: https://commits.webkit.org/295695@main
